modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/fixed/MetadataTypeEnricher.java

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.runtime.module.extension.internal.loader.enricher.fixed;
+
+import static java.util.stream.Collectors.toMap;
+
+import org.mule.metadata.api.annotation.TypeAnnotation;
+import org.mule.metadata.api.builder.BaseTypeBuilder;
+import org.mule.metadata.api.builder.TypeBuilder;
+import org.mule.metadata.api.builder.WithAnnotation;
+import org.mule.metadata.api.model.AnyType;
+import org.mule.metadata.api.model.ArrayType;
+import org.mule.metadata.api.model.AttributeFieldType;
+import org.mule.metadata.api.model.AttributeKeyType;
+import org.mule.metadata.api.model.BinaryType;
+import org.mule.metadata.api.model.BooleanType;
+import org.mule.metadata.api.model.DateTimeType;
+import org.mule.metadata.api.model.DateType;
+import org.mule.metadata.api.model.FunctionType;
+import org.mule.metadata.api.model.LocalDateTimeType;
+import org.mule.metadata.api.model.LocalTimeType;
+import org.mule.metadata.api.model.MetadataType;
+import org.mule.metadata.api.model.NothingType;
+import org.mule.metadata.api.model.NullType;
+import org.mule.metadata.api.model.NumberType;
+import org.mule.metadata.api.model.ObjectFieldType;
+import org.mule.metadata.api.model.ObjectKeyType;
+import org.mule.metadata.api.model.ObjectType;
+import org.mule.metadata.api.model.PeriodType;
+import org.mule.metadata.api.model.RegexType;
+import org.mule.metadata.api.model.StringType;
+import org.mule.metadata.api.model.TimeType;
+import org.mule.metadata.api.model.TimeZoneType;
+import org.mule.metadata.api.model.TupleType;
+import org.mule.metadata.api.model.TypeParameterType;
+import org.mule.metadata.api.model.UnionType;
+import org.mule.metadata.api.model.VoidType;
+import org.mule.metadata.api.model.impl.DefaultAttributeFieldType;
+import org.mule.metadata.api.model.impl.DefaultAttributeKeyType;
+import org.mule.metadata.api.model.impl.DefaultFunctionType;
+import org.mule.metadata.api.model.impl.DefaultObjectFieldType;
+import org.mule.metadata.api.model.impl.DefaultObjectKeyType;
+import org.mule.metadata.api.model.impl.DefaultObjectType;
+import org.mule.metadata.api.model.impl.DefaultUnionType;
+import org.mule.metadata.api.visitor.MetadataTypeVisitor;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+class MetadataTypeEnricher {
+
+  public MetadataType enrich(MetadataType target, Set<TypeAnnotation> annotations) {
```

Comment:
static if the MetadataTypeEnricher has no state, this would be simpler to use as a Util